eip_operator/src/controller/node.rs

```diff
@@ -45,18 +48,33 @@ impl k8s_controller::Context for Context {
         );
 
         let node_ip = node.ip().ok_or(Error::MissingNodeIp)?;
-        let node_labels = node.labels().ok_or(Error::MissingNodeLabels)?;
+        let node_labels = node.labels();
         let provider_id = node.provider_id().ok_or(Error::MissingProviderId)?;
         let instance_id = provider_id
             .rsplit_once('/')
             .ok_or(Error::MalformedProviderId)?
             .1;
-        let all_eips = eip_api.list(&ListParams::default()).await?.items;
-        let eip = all_eips
+        let matched_eips: Vec<Eip> = eip_api
+            .list(&ListParams::default())
+            .await?
+            .items
             .into_iter()
-            .find(|eip| eip.matches_node(node_labels))
-            .ok_or(Error::NoEipResourceWithThatNodeSelector)?;
-        let eip_name = eip.name().ok_or(Error::MissingEipName)?;
+            .filter(|eip| eip.matches_node(node_labels))
+            .collect();
+        if matched_eips.is_empty() {
+            return Err(Error::NoEipResourceWithThatNodeSelector);
+        }
+        let eip = matched_eips.into_iter().find(|eip| {
+            eip.status.as_ref().is_some_and(|s| {
+                s.resource_id.is_none()
+                    || s.resource_id.as_ref().map(|r| r == &name).unwrap_or(false)
+            })
+        });
```

Comment:
How do we handle the migration from older versions of the EIP operator, which don't have resource_id's? Is this OK just because we don't have multiple nodes that match currently?

Reply:
yeah, there could be some re-alignment during this migration if there are multiple nodes matching a single eip, but it shouldn't be any more unstable than the existing implementation.